org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/ui/preferences/XMLCatalogPreferencePage.java

```diff
@@ -0,0 +1,286 @@
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.FileDialog;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPreferencePage;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.ide.IDE;
+import org.eclipse.ui.model.IWorkbenchAdapter;
+import org.eclipse.wildwebdeveloper.xml.internal.Activator;
+import org.eclipse.wildwebdeveloper.xml.internal.ui.Messages;
+
+public class XMLCatalogPreferencePage extends PreferencePage implements IWorkbenchPreferencePage {
+	private TreeViewer viewer;
+	private Set<String> entries;
+	private String selectedEntry;
+	private boolean isDirty;
+	
+	@Override
+	public void init(IWorkbench workbench) {
+		setPreferenceStore(Activator.getDefault().getPreferenceStore());
+	}
+
+	@Override
+	protected Control createContents(Composite parent) {
+		isDirty = false;
+		
+		Composite composite = new Composite(parent, SWT.NONE);
+		GridLayout layout = new GridLayout(1, false);
+		layout.marginWidth = 0;
+		layout.marginHeight = 0;
+		composite.setLayout(layout);
+
+		Group entriesGroup = new Group(composite, SWT.NONE);
+		entriesGroup.setText(Messages.XMLCatalogPreferencesPage_Entries);
+		GridLayout gl = new GridLayout(2, false);
+		entriesGroup.setLayout(gl);
+		entriesGroup.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+		viewer = new TreeViewer(entriesGroup, SWT.H_SCROLL | SWT.V_SCROLL | SWT.BORDER);
+		viewer.setContentProvider(new EntriesContentProvider());
+		viewer.setLabelProvider(new EntriesLabelProvider());
+		
+		entries = XMLPreferenceInitializer.getCatalogs(getPreferenceStore());
+
+		viewer.setInput(entries);
+		viewer.getControl().setLayoutData(new GridData(GridData.FILL_BOTH));
+		viewer.expandAll();
+
+		Composite buttonComposite = new Composite(entriesGroup, SWT.NONE);
+		buttonComposite.setLayout(new GridLayout(1, false));
+		buttonComposite.setLayoutData(new GridData(SWT.FILL, SWT.TOP, false, false));
+
+		Button addButton = new Button(buttonComposite, SWT.PUSH);
+		addButton.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		addButton.setText(Messages.PreferencesPage_Add);
+		addButton.addSelectionListener(widgetSelectedAdapter(e -> {
+			String result = openSelectFileDialog(null);
+			if (result == null || result.trim().length() == 0) {
+				return;
+			}
+			String newEntry = new File(result).getAbsolutePath();
+			entries.add(newEntry);
+			viewer.refresh();
+			viewer.setSelection(new TreeSelection(new TreePath(new String[] {newEntry})));
+			isDirty = true;
+		}));
+		
+		final Button changeButton = new Button(buttonComposite, SWT.PUSH);
```

Comment:
I don't think a "Change" button is worth it, as it's just Remove then Add.
IMO, we should have a "Open in Editor" button instead, that would allow to view and edit the file. That would be much more profitable for users.

Reply:
OK. So I remove`Change` button and rename `Edit` to `Open in Editor` then

Reply:
Yes, remove the Change. However, if you think Edit is a better name, feel free to keep it, I don't have strong opinion here.